src/cml.js

```diff
@@ -159,9 +182,63 @@ class CML {
       ? ''
       : '![CML watermark](https://raw.githubusercontent.com/iterative/cml/master/assets/watermark.svg)';
 
-    const report = `${userReport}\n\n${watermark}`;
+    let userReport = testReport;
+    try {
+      if (!userReport)
+        userReport = await fs.promises.readFile(markdownFile, 'utf-8');
+    } catch (err) {
+      if (!watch) throw err;
+    }
+
+    let report = `${userReport}\n\n${watermark}`;
     const drv = getDriver(this);
 
+    const publishLocalFiles = () => async (tree) => {
+      const nodes = [];
+
+      visit(tree, ['definition', 'image', 'link'], (node) => nodes.push(node));
+
+      const visitor = async (node) => {
+        if (node.url && node.alt !== 'CML watermark') {
+          if (!triggerFile) watcher.add(node.url);
+          try {
+            const url = new URL(
+              await this.publish({ ...opts, path: node.url, session })
+            );
+            url.searchParams.set('cache-bypass', uuid.v4());
+            node.url = url.toString();
+          } catch (err) {
+            if (err.code !== 'ENOENT') throw err;
+          }
+        }
+      };
+
+      await Promise.all(nodes.map(visitor));
+    };
+
+    if (publish)
+      report = String(await remark().use(publishLocalFiles).process(report));
+
+    if (watch) {
+      let lock;
+      watcher.add(triggerFile || markdownFile);
+      watcher.on('all', async (event, path) => {
+        if (lock) return;
+        lock = true;
+        try {
+          winston.info(`watcher event: ${event} ${path}`);
+          await this.commentCreate({ ...opts, update: true, watch: false });
```

Comment:
@0x2b3bfa0 Is there a reason to force `update: true`? Could this respect the `--update` flag?

Reply:
When using `--watch` it always updates the last comment instead of creating a new comment on every file change. Otherwise, it would clutter badly your pull request. 🤔

Reply:
It could respect the update flag, but don't know if it should.

Reply:
> Otherwise, it would clutter badly your pull request

It could be, but there could also be use cases where people want one comment per-update, basically:

> It could honor the update flag, but don't know if it should.

I think we should honor this and let users decide

Reply:
Then, `--update` would be overloaded, meaning both:

#### Meaning 1
Either
* begin the current session by updating the last CML message, or
* begin the current session with a new message

#### Meaning 2
Either
* use a new message for every update, or
* update the same message for every update

Reply:
@0x2b3bfa0 What is the current behavior without the "file-watcher"? 

Not sure I follow why it would be ambiguous when adding the file watcher.

Reply:
Without specifying `--watcher`
* If `--update` is set, it edits the latest CML comment
* If `--update` is not set, it creates a new comment
In this case, the option has a single meaning

Reply:
The question is: should `update` only affect the first run, or also all the subsequent watcher events?

Reply:
Perhaps we should fix https://github.com/iterative/cml/issues/762 first? Something like:

```
cml report {create [--update],watch} <file>
```

Reply:
Would that allow users to “reuse an existing comment and keep updating it”?

Reply:
> _`cml report watch` would imply update-like functionality_

That's the current behavior and the whole point of this thread 😅

Reply:
No, the point of this thread is the API `cml send-comment [--update] --watch <file>` makes no sense, it should be

- `cml report watch <file>`,
- or at least `cml send-comment --watch <file>` should raise
  + `error: --watch needs --update`, or
  + `warning: --watch specified so assuming --update`

Reply:
As per https://github.com/iterative/cml/issues/888#issuecomment-1152312400, an optional `--update` could make sense, sort of

Reply:
> _`cml send-comment --watch <file>` should raise `warning: --watch specified so assuming --update`_

Suspiciously similar to the current behavior, with two exceptions:
- No warning message is being printed
- The `--update` options determines whether to
   - create a new comment and keep updating it, or
   - just use the last CML comment and keep updating it

Reply:
I see. Well #888 seems to be an urgent docs issue, while [forcing update-like behaviour](https://github.com/iterative/cml/pull/1026#discussion_r894391433) seems best to me.

Allowing no-update[^1] breaks the use case IMO.

[^1]: purely as a PAT work-around

Reply:
having `--update` affect the first iteration only (after that - it updates) sounds reasonable, and might be useful. Why does it break the use case @casperdcl ?

Reply:
I think we can leave this for #762, not a blocker for me.

Reply:
resolving unresolved